cmd/channel/email/main.go

```diff
@@ -76,14 +122,35 @@ func (ch *Email) SetConfig(jsonStr json.RawMessage) error {
 			return fmt.Errorf("failed to get the os's current user: %w", err)
 		}
 
-		ch.From = usr.Username + "@" + hostname
+		ch.SenderMail = usr.Username + "@" + hostname
+	}
+
+	if ch.User == "" {
+		ch.User = ch.SenderMail
 	}
 
```

Comment:
Honestly, I am a bit unhappy with all those undocumented fallback or default values.

- `Host` and `Port` might each fall back to `localhost` and `25`, respectively.
  Does this even make sense from a user's perspective? Why should one leave those fields empty anyway?
- `SenderName` is allowed to be left out, resulting in emails with a _From_ header like `From: <${SenderMail}>`, which is totally fine, imho.
- `SenderMail` will fallback to `${OS_USER}@${OS_HOSTNAME}`, which is also quite arbitrary, but something is required.
- `User` and `Password` are being used for the SMTP authentication, but `User` shouldn't fall back to something else, as I mentioned above.
- `Encryption` is the only "required" argument - which, however, is only a web check, relying on this component, and, if being set to neither `starttls` nor `tls`, implicitly falling back to the `none` logic.

Might I suggest not setting any defaults, but marking all necessary fields as "required" and checking if they have valid values in this method, resulting in an error otherwise.